### DISCUSSION TITLE
How to disable scaling for the model / camera zoom disable through user gestures

### DISCUSSION THREAD

**Discussion by v01dn3ph1l1m** [sceneview/sceneview#476] [Q&A]
val engine = rememberEngine()
    val modelLoader = rememberModelLoader(engine)

    val cameraNode = rememberCameraNode(engine) {
        isScaleEditable = false
    }
    val centerNode = rememberNode(engine)
        .addChildNode(cameraNode)
    val modelInstance = remember { modelLoader.createModelInstance("female_mode.glb") }

    val modelNode = rememberNode {
        ModelNode(
            modelInstance = modelInstance,
            scaleToUnits = 0.4f,
            autoAnimate = false
        )

    }.apply {
        isScaleEditable = false
    }
    
    
            Scene(
        modifier = modifier.fillMaxSize(),
        engine = engine,
        modelLoader = modelLoader,
        cameraNode = cameraNode,
        isOpaque = false,
        childNodes = listOf(centerNode, modelNode),
        onFrame = {
            cameraNode.lookAt(centerNode)
        }
    )
i tried this but didnt work

also tried this

   val onGestureListener = rememberOnGestureListener(
       onScaleBegin = { _, _, _ -> } 
        onScale = { _, _, _ -> } 
    )

I am on version 2.1.1 and latest version of jetpack compose

I also tried setting isScaleEditable on cameraNode, centerNode and modelNode but user gestures still work and scale was still editable. Can someone help?

**Comment by v01dn3ph1l1m** [comment]
@ThomasGorisse stuck here. Would appreciate some guidance in the right direction.

**Reply by 1729patrick** [reply]
@v01dn3ph1l1m have you figured out how to disable gestures on the camera? Thanks!

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, you can configure gesture handling through the `cameraManipulator`:

```kotlin
SceneView(
    cameraManipulator = rememberCameraManipulator(
        // Disable zoom by setting scroll limits
    ),
    // Or disable all camera gestures:
    gestureDetector = null
) {
    ModelNode(modelInstance = model)
}
```

For AR scenes, you can control the gesture detector on `ARSceneView` to disable specific interactions. You can also set `scaleToUnits` on `ModelNode` to fix the model size and prevent user-initiated scaling.
